core.spec.md

```diff
@@ -0,0 +1,501 @@
+    - If {name} exists within {features}, the ***Name Uniqueness* validation fails**.
+    - Insert {name} => {d} into {features}
+  - **Return** {features}
+
+
+Prefixes, whether implicit or explicit, must be unique within a document. Valid:
+
+:::[example](prefixing.graphql#schema[0]) -- Unique prefixes
+
+It is also valid to reference multiple versions of the same spec under different prefixes:
+
+:::[example](prefix-uniqueness.graphql#schema[0]) -- Explicit prefixes allow multiple versions of the same spec to coexist within a Document
+
+Without the explicit [`as:`](#@core/as), the above would be invalid:
+
+:::[counter-example](prefix-uniqueness.graphql#schema[1]) -- Non-unique prefixes with multiple versions of the same spec
+
+Different specs with the same prefix are also invalid:
+
+:::[counter-example](prefix-uniqueness.graphql#schema[2]) -- Different specs with non-unique prefixes
+
+## Assign Features
+
+Create a map of {element}: *Any Named Element* -> {feature}: `Directive` | {null}, associating every named schema element within the document with a feature directive, or {null} if it is not associated with a feature.
+
+AssignFeatures(document) :
+  - Let {features} be the result of collecting features via {CollectFeatures(document)}
+  - Let {assignments} be a map of ({element}: *Any Named Element*) -> {feature}: `Directive` | {null}, initally empty
+  - For each named schema element {e} within the {document}
+    - Let {name} be the name of the {e}
+    - If {name} begins with {"__"},
+      - Insert {e} => {null} into {assignments}
+      - **Continue** to next {e}
+    - If {name} contains the substring {"__"},
+      - Partition {name} into `[`{prefix}, {base}`]` at the first {"__"} (that is, find the shortest {prefix} and longest {base} such that {name} = {prefix} + {"__"} + {base})
+      - If {prefix} exists within {features}, insert {e} => {features}`[`{prefix}`]` into {assignments}
+        - Else, insert {e} => {null} into {assignments}
+      - **Continue** to next {e}
+    - Insert `e => null` into {assignments}
+  - **Return** {assignments}
+
+## Is Exported?
+
+Determine if any schema element is [exported](#sec-Parts-of-a-Core-Schema) to the API. A core schema's [API](#sec-Parts-of-a-Core-Schema) is always the subset of the entire Document containing only exported elements.
+
+IsExported(element) :
+  - Let {coreName} be the name of the core feature found via {Bootstrap(document)}
+  - Let {assignments} be the result of assigning features to elements via {AssignFeatures(document)}
+  - For each Directive {d} on {element},
+    - If {d}'s name is {coreName}`__export`,
+      - If {d} does not have an `export:` argument *or* `export:` is {true}, **Return** {true}
+      - If {d} has an `export:` argument whose value is {false}, **Return** {false}
```

Comment:
CLARITY: The wording here is inconsistent with the previous line ("`export:` is true" vs "`export:` argument whose value is false")